homeassistant/components/lovelace/__init__.py

```diff
@@ -1,6 +1,7 @@
 """Lovelace UI."""
 import logging
 import uuid
+import time
```

Comment:
We should not have a time based approach. I think that it's fine to always cache the last loaded response. When a user clicks on refresh in the Lovelace config, we can send `force_refresh: true` and then load it again from file.

Reply:
But then we risk overwriting edits users have done manual in ui-lovelace.yaml, because hass is not aware of them.

Reply:
on save, we can load the latest version and update that?

Reply:
And if the user ends up in an invalid state that does not apply to the latest config because of writes both local and remote, it's better that we fail the update

Reply:
(ID not found etc)

Reply:
We could, but would be less efficient... What is against using time?

Reply:
What would be less efficient?

I propose something like this:

- Normal load -> we use cached one, load if nothing cached
- Force load -> we clear cache and do normal load
- Save -> we load latest file, update it, write it and store in cache

Reply:
We discussed on Discord, Bram is right. Just need to migrate to `hass.data` instead of global.